### PR TITLE
Improved carrier paths

### PR DESCRIPTION
**Pull request by Kurtzmusch**
option between thin or dashed looped paths
configurable path width
path scales with clamped scale setting

### CODE REVIEW

File: client/src/components/game/menu/Options.vue [hunk 1]
```diff
@@ -104,6 +104,34 @@
...
+          <input type="number" class="form-control" id="carrier-path-width" v-model="settings.map.carrierPathWidth" :disabled="isSavingSettings">
+        </div>
+      </div>
+
+      <div  class="row bg-secondary pt-1 pb-1">
+        <label for="loop-style" class="col-12 col-sm-6 col-form-label">Loop Style</label>
+        <div class="col-12 col-sm-6">
+          <select class="form-control" id="loop-style" v-model="settings.map.carrierLoopStyle" :disabled="isSavingSettings">
+            <option value="dashed">Dashed</option>
+            <option value="thin">Thin</option>
```
[the remark this answers is not in the document]
Comment (Kurtzmusch) [reply client/src/components/game/menu/Options.vue:123]: sure, often times i don't really come up with good names for these things and just use whatever comes to mind

File: client/src/game/carrier.js [hunk 2]
```diff
@@ -209,27 +211,167 @@ class Carrier extends EventEmitter {
...
+      path.lineTo(Math.min(initialX+(dashCount*COMBINED_LENGTH),pathLength), -lineWidth)
+      path.lineTo(pathLength, -lineWidth)
+      path.lineTo(pathLength, lineWidth)
+      path.endFill()
+
+      path.rotation = Math.atan2(pointB.y-pointA.y,pointB.x-pointA.x)
+      path.position = pointA
+      this.pathContainer.addChild(path)
+
+      //TODO make line caps optional since they are barely visible and shit performance
```
[the remark this answers is not in the document]
Comment (Kurtzmusch) [reply client/src/game/carrier.js:270]: yes, keep it

File: client/src/game/carrier.js [hunk 3]
```diff
@@ -209,27 +211,167 @@ class Carrier extends EventEmitter {
...
+        this._drawRegularPathSegment(lineWidth, lineAlpha, lastPoint, star)
+      }
+
+      lastPoint = star
+    }
+    //draw path back to the first destination
+    if ( this.data.waypointsLooped ) {
+      if ( !sourceIsLastDestination ) {
+        let firstPoint
+        firstPoint = this.stars.find(s => s.data._id === this.data.waypoints[0].destination)
```
[the remark this answers is not in the document]
Comment (Kurtzmusch) [reply client/src/game/carrier.js:370]: OK, i'll do that

[the remark this answers is not in the document]
Comment (Kurtzmusch) [reply client/src/game/carrier.js:370]: Actually no, i need the mapObject for the line caps, and the gameHelper returns a mapObject.data. I could write another helper function on the gameHelper though.

[the remark this answers is not in the document]
Comment (mike-eason) [reply client/src/game/carrier.js:370]: Nah that's cool as is.

**Pull request merged**